pgqueuer/qm.py

```diff
@@ -386,27 +408,35 @@ async def _dispatch(
             job.id,
         )
 
-        async with self.entrypoint_statistics[job.entrypoint].concurrency_limiter:
+        async with (
+            heartbeat.Heartbeat(
+                job.id,
+                self.entrypoint_registry[job.entrypoint].retry_timer,
```

Comment:
Does having the heartbeat interval be the same as the `retry_timer` cause heartbeats to race with the retry? Perhaps it would be more reliable if the heartbeat interval was `retry_timer / 2` by default, or configurable by the entrypoint?

Reply:
Well spotted, i had a / 2 at some point, must have lost it during a refactor.